chips/nrf52/src/spi.rs

```diff
@@ -190,9 +190,22 @@ pub enum Frequency {
     M8 = 0x80000000,
 }
 
-impl From<Frequency> for u32 {
-    fn from(freq: Frequency) -> u32 {
-        match freq {
+impl Frequency {
+    pub fn from_register(reg: u32) -> Option<Frequency> {
```

Comment:
why not implement `TryFrom` instead?

Reply:
I removed all `From<>` trait implementations in an effort to avoid confusion. Implementing `TryFrom<u32> for Frequency` again doesn't clearly state, whether we're talking about register value or frequency.

Reply:
Ok, but in this case it is the frequency value, no?

Reply:
Ok, I get your point